tools/lou_checkyaml.c

```diff
@@ -813,30 +816,39 @@ read_test(yaml_parser_t *parser, char **tables, int direction, int hyphenation)
 	int result = 0;
 	char **table = tables;
 	while (*table) {
+		int r;
 		if (hyphenation == HYPHENATION_ON) {
-			result |= check_hyphenation(*table, word, translation);
+			r = check_hyphenation(*table, word, translation);
 		} else {
 			// FIXME: Note that the typeform array was constructed using the
 			// emphasis classes mapping of the last compiled table. This
 			// means that if we are testing multiple tables at the same time
 			// they must have the same mapping (i.e. the emphasis classes
 			// must be defined in the same order).
-			result |= check(*table, word, translation, .typeform = typeform, .mode = mode,
+			r = check(*table, word, translation, .typeform = typeform, .mode = mode,
 					.expected_inputPos = inPos, .expected_outputPos = outPos,
 					.cursorPos = cursorPos, .expected_cursorPos = cursorOutPos,
 					.max_outlen = maxOutputLen, .real_inlen = realInputLen,
 					.direction = direction, .diagnostics = !xfail);
 		}
+		if (xfail != r) errors++;
+		if ((xfail || r != 0) && !(verbose == 0 && xfail && r != 0)) {
+			if (description) fprintf(stderr, "%s\n", description);
```

Comment:
This is really hard to understand. I'd even prefer something non-dry like
``` C
if (xfail != r) {
	if (description) fprintf(stderr, "%s\n", description);
	error_at_line(0, 0, file_name, event.start_mark.line + 1,
			(xfail ? "Unexpected Pass" : "Failure"));
	errors++;
}
if (xfail && r && verbose) {
        // in verbose mode print expected failures
	if (description) fprintf(stderr, "%s\n", description);
	error_at_line(0, 0, file_name, event.start_mark.line + 1, "Expected Failure");
}
```

Reply:
Yeah, much easier to read. The `&& r` is not even needed. I would make it

```c
if (xfail != r) {
    ...
} else if (xfail && verbose) {
    ...
}
```

Reply:
Hm, that makes it harder to read again IMHO, because then I have to mentally reconstruct that we are only interested in cases where the test failed (r because !(xfail != r)) that are expected failures (xfail) and we are in verbose mode (verbose)

Reply:
You can never have a expected failure AND a unexpected pass or failure, hence the `else if`. It's easy to understand.

You can argue about whether the `&& r` is needed or not. I thought because of the `else if` it's easy to deduct, but I can leave it if you want.